scep/certs_selector.go

```diff
@@ -0,0 +1,30 @@
+package scep
+
+import "crypto/x509"
+
+// A CertsSelector filters certificates.
+type CertsSelector interface {
+	SelectCerts([]*x509.Certificate) []*x509.Certificate
+}
+
+// NopCertsSelector is a CertsSelector that does not do anything.
+type NopCertsSelector struct{}
```

Comment:
Just a nit, but I kinda like the `HandlerFunc` pattern over the empty structs. I.e.:

```
type CertsSelectorFunc func([]*x509.Certificate) []*x509.Certificate

func (f *CertsSelectorFunc) SelectCerts(certs []*x509.Certificate) []*x509.Certificate {
    return f(certs)
}
```

And then NopSelector becomes:

```
func NopCertsSelector() CertsSelectorFunc {
    return func(certs []*x509.Certificate) []*x509.Certificate {
        return certs
    }
}
```

Thoughts?

Reply:
Done
I am still super new to Go, so you may find me adapting best practices of other languages. Thanks for teaching me the Go way of doing things :-)